### PR TITLE
Microsoft Clarity destination docs

### PR DESCRIPTION
**Pull request by ameypv-rudder**
## What do these changes do?

> Added documentation for the Microsoft Clarity destination.

## Type of documentation

- [x] New documentation
- [ ] Documentation update
- [ ] Hotfix

### CODE REVIEW

File: docs/destinations/streaming-destinations/microsoft-clarity.mdx [hunk 1]
```diff
@@ -0,0 +1,79 @@
+---
+title: "Microsoft Clarity"
+description: Step-by-step guide on sending your event data from RudderStack to Microsoft Clarity.
+aliases: ["Microsoft Clarity destination"]
+---
+
+[Microsoft Clarity](https://clarity.microsoft.com/) is a heatmaps and session recording tool. It helps you better understand how users are interacting with your website through session replays and heatmaps.
+
+RudderStack supports Microsoft Clarity as a destination where you can seamlessly send your event data.
```
Comment (ishachopra0401) [inline docs/destinations/streaming-destinations/microsoft-clarity.mdx:9]:
```suggestion
```

can skip according to our latest destination template

File: docs/destinations/streaming-destinations/microsoft-clarity.mdx [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+
+To successfully configure Microsoft Clarity as a destination, you will need to configure the following settings:
+
+- **Project ID**: Enter your Microsoft Clarity project ID. For more information on obtaining your Clarity project ID, refer to the <Link to="#faq">FAQ</Link> section below.
+- **Microsoft Clarity Cookie Consent**: Enable this setting to allow RudderStack to call the Microsoft Clarity API to track the users using a cookie.
+- **Client-side Events Filtering**: This setting lets you specify which events should be blocked or allowed to flow through to Microsoft Clarity. For more information on this setting, refer to the <Link to="/sources/sdks/event-filtering/">Client-side Events Filtering</Link> guide.
+- **Use device mode to send events**: As this is a <Link to="https://www.rudderstack.com/docs/destinations/rudderstack-connection-modes/#device-mode">web device mode</Link>-only destination, this setting is enabled by default and cannot be disabled.
+- - **OneTrust Cookie Categories**: This setting lets you associate <Link to="/sources/sdks/rudderstack-javascript-sdk/consent-managers/onetrust/">OneTrust</Link> cookie consent groups to Microsoft Clarity.
```
Comment (ishachopra0401) [inline docs/destinations/streaming-destinations/microsoft-clarity.mdx:33]:
```suggestion
- **OneTrust Cookie Categories**: This setting lets you associate <Link to="/sources/sdks/rudderstack-javascript-sdk/consent-managers/onetrust/">OneTrust</Link> cookie consent groups to Microsoft Clarity.
```

File: docs/destinations/streaming-destinations/microsoft-clarity.mdx [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+A sample `identify` call is shown below:
+
+```javascript
+rudderanalytics.identify("1hKOmRA4GRlm", {
+  name: "Alex Keener",
+  email: "alex@example.com"
+});
+```
+
+By default, Clarity uses cookies to track your users and gather the session data. You can configure your Clarity project to require <a href="https://learn.microsoft.com/en-us/clarity/cookie-consent">cookie consent</a>, that is, Clarity **will not** place a cookie until the user's consent is obtained.
```
Comment (ishachopra0401) [inline docs/destinations/streaming-destinations/microsoft-clarity.mdx:48]:
```suggestion
By default, Clarity uses cookies to track your users and gather the session data. You can configure your Clarity project to require <a href="https://learn.microsoft.com/en-us/clarity/cookie-consent">cookie consent</a>, that is, Clarity **will not** place a cookie until a user provides consent.
```

File: docs/destinations/streaming-destinations/microsoft-clarity.mdx [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+
+<img src="../../assets/event-stream-destinations/clarity-cookie-consent.png" alt="Microsoft Clarity cookie consent enabled" />
+
+<br />
+
+In this case, to track the user sessions via the cookie, make sure you enable the **Microsoft Clarity Cookie Consent** setting in the RudderStack dashboard. This setting indicates that RudderStack can call the Clarity API to track the users via a cookie once the consent is obtained.
```
Comment (ishachopra0401) [inline docs/destinations/streaming-destinations/microsoft-clarity.mdx:54]:
```suggestion
In this case, to track the user sessions via the cookie, make sure you enable the **Microsoft Clarity Cookie Consent** setting in the RudderStack dashboard. This setting indicates that RudderStack can call the Clarity API to track the users via a cookie once a user provides consent.
```

**Review by ishachopra0401 — APPROVED** [review]

**Comment by ameypv-rudder** [comment]
@mihir-4116 @shrouti1507 could you please have a look at the PR and let me know if we need to change/tweak anything?

**Review by mihir-4116 — APPROVED** [review]

**Pull request merged**